rpc/wrtc_client.go

```diff
@@ -230,37 +218,51 @@ func dialWebRTC(
 		}
 
 		var pendingCandidates sync.WaitGroup
+
+		// waitOneHost is closed when the first ICE candidate of type `Host` (e.g: 127.0.0.1) is
+		// found.
 		waitOneHost := make(chan struct{})
 		var waitOneHostOnce sync.Once
 		peerConn.OnICECandidate(func(icecandidate *webrtc.ICECandidate) {
 			if exchangeCtx.Err() != nil {
+				// We've decided to bail.
 				return
 			}
+
 			if icecandidate != nil {
+				// The last `icecandidate` called from pion will be nil. `nil` signifies that all
+				// candidates were created. We will still create a goroutine for this "empty"
+				// candidate to wait for all other candidates to complete. Thus we only increment
+				// `pendingCandidates` for non-nil values.
 				pendingCandidates.Add(1)
 				if icecandidate.Typ == webrtc.ICECandidateTypeHost {
 					waitOneHostOnce.Do(func() {
 						close(waitOneHost)
 					})
 				}
 			}
+
 			// must spin off to unblock the ICE gatherer
 			utils.PanicCapturingGo(func() {
 				if icecandidate != nil {
 					defer pendingCandidates.Done()
 				}
 				select {
 				case <-remoteDescSet:
+					// We've sent the `init` offer. We can now proceed with sending individual
```

Comment:
I thought `remoteDescSet` gets closed when we've _received_ an init offer from the answerer in a `CallResponse_Init`.
```suggestion
					// We've received the `init` offer. We can now proceed with sending individual
```

Reply:
I think there's two issues here:
* I chose words poorly -- it's not important that we sent the offer, but received the "init response".
* While that response does contain the "answer" (not the "offer") -- the thing this function actually cares about is the "exchange UUID". It's legal to [read this variable](https://github.com/viamrobotics/goutils/blob/e8de5415d7fc5848e5d7a80e62c04f4fa2f38154/rpc/wrtc_client.go?plain=1#L267) because we've observed the channel being closed (i.e: in happens-before relationship notation, the uuid being written to happens before the channel is closed).

Reply:
Yeah good clarification about the UUID; what you have now makes sense to me.